sumologic/sumologic_muting_schedules_library_muting_schedule.go

```diff
@@ -0,0 +1,173 @@
+package sumologic
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// ---------- ENDPOINTS ----------
+
+func (s *Client) CreateMutingSchedulesLibraryMutingSchedule(mutingSchedulesLibraryMutingSchedule MutingSchedulesLibraryMutingSchedule, paramMap map[string]string) (string, error) {
+	urlWithoutParams := "v1/mutingSchedules"
+	paramString := ""
+	sprintfArgs := []interface{}{}
+
+	paramString += "?"
+
+	if val, ok := paramMap["parentId"]; ok {
+		queryParam := fmt.Sprintf("parentId=%s&", val)
+		paramString += queryParam
+	}
+
+	urlWithParams := fmt.Sprintf(urlWithoutParams+paramString, sprintfArgs...)
+
+	data, err := s.Post(urlWithParams, mutingSchedulesLibraryMutingSchedule)
+	if err != nil {
+		return "", err
+	}
+
+	var createdMutingSchedulesLibraryMutingSchedule MutingSchedulesLibraryMutingSchedule
+
+	err = json.Unmarshal(data, &createdMutingSchedulesLibraryMutingSchedule)
+	if err != nil {
+		return "", err
+	}
+
+	return createdMutingSchedulesLibraryMutingSchedule.ID, nil
+
+}
+
+func (s *Client) MutingSchedulesRead(id string) (*MutingSchedulesLibraryMutingSchedule, error) {
+	urlWithoutParams := "v1/mutingSchedules/%s"
+	paramString := ""
+	sprintfArgs := []interface{}{}
+	sprintfArgs = append(sprintfArgs, id)
+
+	urlWithParams := fmt.Sprintf(urlWithoutParams+paramString, sprintfArgs...)
+
+	data, _, err := s.Get(urlWithParams)
+	if err != nil {
+		return nil, err
+	}
+	if data == nil {
+		return nil, nil
+	}
+
+	var mutingSchedulesLibraryMutingSchedule MutingSchedulesLibraryMutingSchedule
+
+	err = json.Unmarshal(data, &mutingSchedulesLibraryMutingSchedule)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &mutingSchedulesLibraryMutingSchedule, nil
+}
+
+func (s *Client) DeleteMutingSchedulesLibraryMutingSchedule(id string) error {
+	urlWithoutParams := "v1/mutingSchedules/%s"
+	paramString := ""
+	sprintfArgs := []interface{}{}
+	sprintfArgs = append(sprintfArgs, id)
+
+	urlWithParams := fmt.Sprintf(urlWithoutParams+paramString, sprintfArgs...)
+
+	_, err := s.Delete(urlWithParams)
+
+	return err
+}
+
+func (s *Client) UpdateMutingSchedulesLibraryMutingSchedule(mutingSchedulesLibraryMutingSchedule MutingSchedulesLibraryMutingSchedule) error {
+	urlWithoutParams := "v1/mutingSchedules/%s"
+	paramString := ""
+	sprintfArgs := []interface{}{}
+	sprintfArgs = append(sprintfArgs, mutingSchedulesLibraryMutingSchedule.ID)
+
+	urlWithParams := fmt.Sprintf(urlWithoutParams+paramString, sprintfArgs...)
+
+	mutingSchedulesLibraryMutingSchedule.ID = ""
```

Comment:
curious why we are setting the value as empty here?

Reply:
I refer to the code from here https://github.com/SumoLogic/terraform-provider-sumologic/blob/77790f828653d88601bdb8f0e3bcb657d2213cef/sumologic/sumologic_monitors_library_monitor.go#L88
And though it may be a required operation.

Reply:
Evaluate and remove it.

Reply:
yes. I removed it.